internal/operands/common-instancetypes/reconcile_test.go

```diff
@@ -165,6 +171,93 @@ var _ = Describe("Common-Instancetypes operand", func() {
+		assertResoucesExist(request, virtualMachineClusterInstancetypes, virtualMachineClusterPreferences)
+
+		// Assert that CrdWatch can see the required CRDs before we call Cleanup
+		Expect(request.CrdWatch.CrdExists(virtualMachineClusterInstancetypeCrd)).To(BeTrue())
+		Expect(request.CrdWatch.CrdExists(virtualMachineClusterPreferenceCrd)).To(BeTrue())
+
+		_, err = operand.Cleanup(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		assertResoucesNotExist(request, virtualMachineClusterInstancetypes, virtualMachineClusterPreferences)
+	})
+
+	It("should cleanup resources no longer provided after switching from the bundle to URL", func() {
+		// Reconcile from the internal bundle including both VirtualMachineClusterPreferences and VirtualMachineClusterInstancetypes
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		virtualMachineClusterInstancetypesFromBundle, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterInstancetype](instancetypePath)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(virtualMachineClusterInstancetypesFromBundle).ToNot(BeEmpty())
+
+		virtualMachineClusterPreferencesFromBundle, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterPreference](preferencePath)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(virtualMachineClusterPreferencesFromBundle).ToNot(BeEmpty())
+		assertResoucesExist(request, virtualMachineClusterInstancetypesFromBundle, virtualMachineClusterPreferencesFromBundle)
+
+		// Switch over to the URL and only install VirtualMachineClusterPreferences from common-instancetypes
+		request.Instance.Spec.CommonInstancetypes = &ssp.CommonInstancetypes{
+			URL: pointer.String("https://github.com/kubevirt/common-instancetypes//VirtualMachineClusterPreferences"),
+		}
+		_, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Assert that that all VirtualMachineClusterInstancetypes have been cleaned up
+		// We can't use virtualMachineClusterPreferencesFromBundle here as these resources are also provided by the URL
+		assertResoucesNotExist(request, virtualMachineClusterInstancetypesFromBundle, nil)
+
+		// Assert that that the expected VirtualMachineClusterPreferences are present
+		virtualMachineClusterInstancetypesFromURL, virtualMachineClusterPreferencesFromURL, err := FetchResourcesFromURL(*request.Instance.Spec.CommonInstancetypes.URL)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(virtualMachineClusterInstancetypesFromURL).To(BeEmpty())
+		Expect(virtualMachineClusterPreferencesFromURL).ToNot(BeEmpty())
+		assertResoucesExist(request, virtualMachineClusterInstancetypesFromURL, virtualMachineClusterPreferencesFromURL)
+
+	})
+
+	It("should cleanup resources no longer provided after updating the URL", func() {
+		request.Instance.Spec.CommonInstancetypes = &ssp.CommonInstancetypes{
+			URL: pointer.String("https://github.com/kubevirt/common-instancetypes//VirtualMachineClusterPreferences"),
```

Comment:
Can this work without ref if the validation webhook denies URL without ref?

Reply:
We don't exercise the webhooks from these unit tests so yes it works.